File: backend/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel


class CriterionResponse(BaseModel):
    id: int
    nos_unit_id: int
    criterion_code: str
    criterion_text: str
    page_reference: str | None
    order_index: int

    model_config = {"from_attributes": True}


class NOSUnitResponse(BaseModel):
    id: int
    document_id: int
    unit_code: str
    unit_title: str
    description: str | None
    order_index: int
    criteria: list[CriterionResponse] = []

    model_config = {"from_attributes": True}


class DocumentResponse(BaseModel):
    id: int
    user_id: int
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    status: str
    error_message: str | None
    uploaded_at: datetime
    processed_at: datetime | None

    model_config = {"from_attributes": True}


class DocumentListResponse(BaseModel):
    documents: list[DocumentResponse]
    total: int
    page: int
    per_page: int

File: frontend/src/types/document.ts
export interface Document {
  id: number;
  user_id: number;
  filename: string;
  original_filename: string;
  file_type: 'pdf' | 'docx' | 'txt';
  file_size: number;
  status: 'uploaded' | 'processing' | 'analyzed' | 'failed';
  error_message: string | null;
  uploaded_at: string;
  processed_at: string | null;
}

export interface NOSUnit {
  id: number;
  document_id: number;
  unit_code: string;
  unit_title: string;
  description: string | null;
  order_index: number;
  criteria: PerformanceCriterion[];
}

export interface PerformanceCriterion {
  id: number;
  nos_unit_id: number;
  criterion_code: string;
  criterion_text: string;
  page_reference: string | null;
  order_index: number;
}
